From 3114f3c042d4484fbc066f1f75a4700039df13d5 Mon Sep 17 00:00:00 2001
From: Lionel Ulmer <lionel.ulmer@intel.com>
Date: Tue, 29 Sep 2015 17:49:07 +0200
Subject: [PATCH] tty: fix TTY reference leak

A reference to the TTY was taken before scheduling the hangup
work but was not released right away if this scheduling
failed.

Fix the issue by releasing the acquired reference right away
in case 'schedule_work' returns false.

Change-Id: I18e2b78b5a859177c1141189655e80d4152425a3
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-6611
Signed-off-by: Lionel Ulmer <lionel.ulmer@intel.com>
---
 drivers/tty/tty_io.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/tty_io.c b/drivers/tty/tty_io.c
index b86551c..6cc4e35 100644
--- a/drivers/tty/tty_io.c
+++ b/drivers/tty/tty_io.c
@@ -731,7 +731,8 @@ void tty_hangup(struct tty_struct *tty)
 	printk(KERN_DEBUG "%s hangup...\n", tty_name(tty, buf));
 #endif
 	tty_kref_get(tty);
-	schedule_work(&tty->hangup_work);
+	if (!schedule_work(&tty->hangup_work))
+		tty_kref_put(tty);
 }
 
 EXPORT_SYMBOL(tty_hangup);
-- 
1.9.1

